정민지/mvvmstudy/app/src/main/java/com/github/sookhee/mvvmstudy/interactor/MainInteractor.kt

```diff
@@ -0,0 +1,81 @@
+package com.github.sookhee.mvvmstudy.interactor
+
+import android.util.Log
+import com.github.sookhee.mvvmstudy.network.spec.GithubRepositoryResponse
+import com.github.sookhee.mvvmstudy.network.GithubAPI
+import com.github.sookhee.mvvmstudy.network.RetrofitClient
+import com.github.sookhee.mvvmstudy.network.spec.GithubRepositoryListResponse
+import com.github.sookhee.mvvmstudy.ui.main.MainContract
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+/**
+ *  MainInteractor.kt
+ *
+ *  Created by Minji Jeong on 2022/04/25
+ *  Copyright © 2022 MashUp All rights reserved.
+ */
+
+class MainInteractor : MainContract.Interactor {
+    override fun getGithubRepositoryList(onNetworkCallbackListener: MainContract.Interactor.OnNetworkCallbackListener) {
+        val request = RetrofitClient.buildService(GithubAPI::class.java)
+        val call = request.getRepository()
+
+        call.enqueue(object : Callback<List<GithubRepositoryResponse>> {
+            override fun onResponse(
+                call: Call<List<GithubRepositoryResponse>>,
+                response: Response<List<GithubRepositoryResponse>>
+            ) {
+                Log.d(TAG, "onResponse: $response")
+
+                if (response.isSuccessful) {
+                    Log.d(TAG, "onResponse: ${response.body()}")
+                    response.body()?.let {
+                        onNetworkCallbackListener.onSuccess(it)
+                    }
+                } else if (response.code() != 200) {
+                    onNetworkCallbackListener.onFailure(Throwable("${response.code()}"))
+                }
```

Comment:
혹시 else if로 체크하신 이유가 있을까요?
isSuucessful 내부 구현이 아래처럼 되어 있어서 혹시 에러를 구분하기 위함이라면 else로만 두어도 의도하신 대로 동작해보이네요!

```
   val isSuccessful: Boolean
    get() = code in 200..299
```    
   
   
```suggestion
                } else {
                    onNetworkCallbackListener.onFailure(Throwable("${response.code()}"))
                }
```

Reply:
오 내부 구현까지..! 감사합니다 참고해서 수정할게요!